### PR TITLE
Restructuring authentication on portal

### PR DESCRIPTION
**Pull request by frimpongopoku**
Technically, you are not meant to see any changes in functionality on the portal. 
This arrangement in authentication was to make the whole process make sense to anyone(devs), and allow easy manipulations when we want to make changes. 

@BradHN1  please leave all concerns here, I will correct them by the time you wake up tomorrow. 

Everything "authentication" should be working as it should. We do need to test everything manually until I write the tests. I have tested manually though, all is working well; but I am sure thats a scam 🤣 🤣  Lets do it again, thanks.

### CODE REVIEW

**Comment by BradHN1** [comment]
Testing, noted that when I select the Google or Facebook button on sign in on the passwordless page, nothing happens
Google signin works from the email+password page

File: src/components/Pages/Auth/shared/components/ErrorBoard.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
```
Comment (BradHN1) [inline src/components/Pages/Auth/shared/components/ErrorBoard.js:1]: Not sure what this file is for, did you mean to commit it?

File: src/components/Pages/ProfilePage/DeleteAccountForm.js [hunk 2]
```diff
@@ -113,65 +119,48 @@ class DeleteAccountFormBase extends React.Component {
...
-            .currentUser.delete()
-            .then(() => {
-              apiCall("users.delete", { user_id: this.props.user.id }).then(
-                (json) => {
-                  this.props.firebase.auth().signOut();
-                  this.props.reduxLogout();
-                }
-              );
-            });
+      Auth.currentUser.reauthenticateWithCredential(cred).then(() => {
```
Comment (BradHN1) [inline src/components/Pages/ProfilePage/DeleteAccountForm.js:122]: I just changed this in my BHN-fixes branch.  Need to make sure this works for passwordless.  Do we need to reauthenticate to delete the account?

File: src/components/Pages/Auth/AuthEntry.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      });
+    setLoading(true);
+    firebaseRegistration(form, () => setLoading(false));
+  };
+
+  // ---------------------------------------------------------------
+
+  if (fireAuth && !fireAuth.emailVerified) return <VerifyEmailBox />;
+
+  if (userIsAuthenticated) return <Redirect to={links.profile} />;
```
Comment (Davidjwkinghorn) [inline src/components/Pages/Auth/AuthEntry.js:86]: I think we should redirect to the home page. @BradHN1 what do you think?

**Comment by BradHN1** [comment]
Good point.  Or redirect to the page which directed the user to sign in, could be a team page or action page.

**Review by Davidjwkinghorn — APPROVED** [review]

**Comment by Davidjwkinghorn** [comment]
That makes sense either would be good.

**Comment by BradHN1** [comment]
I'm testing the account deletion with the passwordless authentication.  When I click the link "Click here to send..." nothing is sent to my email, and I get a couple errors in the javascript console:
Screen Shot 2022-02-03 at 6 15 39 AM

**Comment by frimpongopoku** [comment]
@BradHN1 , deleting profile for: 

- [X] Email and password 
- [X] PasswordFree 
- [X] Google Auth 
- [X] Facebook Auth 

All should work now.
-[X] Also if you are signed in, you will get redirected to your last known location

**Review by BradHN1 — APPROVED** [review]
Good work, let's test in canary

**Pull request merged**